src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -248,10 +245,59 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             StartingDirectoryUseParentCheckbox().IsChecked(true);
         }
 
+        // Subscribe to some changes in the view model
+        // These changes should force us to update our own set of "Current<Setting>" members,
+        // and propagate those changes to the UI
+        _ViewModelChangedRevoker = _State.Profile().PropertyChanged(winrt::auto_revoke, [=](auto&&, const PropertyChangedEventArgs& args) {
+            if (args.PropertyName() == L"CursorShape")
```

Comment:
minor opportunity for optimization: pull the property name out into an auto-typed local. You can avoid making more and more calls through the projected interface to get it.

Reply:
Done. Did it in `ProfileViewModel`'s `PropertyChanged` handler too. Should I also pull out the strings into `static constexpr std::wstring_view`s? Like this:
```c++
static constexpr std::wstring_view CursorShapeKey{ L"CursorShape" };
static constexpr std::wstring_view CurrentCursorShapeKey{ L"CurrentCursorShape" };
static constexpr std::wstring_view IsVintageCursorKey{ L"IsVintageCursor" };
...
```

Reply:
Doing that is additional clutter for no real benefit :)